include/materials/czm/CZMMaterialBasePK1.h

```diff
@@ -0,0 +1,259 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InterfaceMaterial.h"
+
+/**
+ * This is the base Material class for implementing a traction separation
+ * material model including large rotation and area changes. The responsibility
+ * of this class is to rotate the displacement jump from global to local
+ * coordinate and rotate back traction and traction derivatives. The local
+ * coordinate system assumes the following order: opening, tangential1,
+ * tangential2. Note that tangential1, tangential2 are arbitrary and therefore
+ * the interface assumes an in-plane isotropic behavior. By overriding
+ * computeTractionIncrementAndDerivatives in aderived class, different traction
+ * separation laws can be implemented. The
+ * computeTractionIncrementAndDerivatives method assumes calculations are
+ * performed in the local frame. CZM laws should always be implemented in 3D
+ * even if they are going to be used in 2D or 1D simulations. This class assumes
+ * large rotation, large area changes, and that the traction separation law is
+ * only dependent upon the the displacement jump.
+ */
+class CZMMaterialBasePK1 : public InterfaceMaterial {
+
+public:
+  static InputParameters validParams();
+  CZMMaterialBasePK1(const InputParameters &parameters);
+
+protected:
+  virtual void initialSetup() override;
+
+  /// this method is responsible for computing the traction incremenet  and its
+  /// derivatives w.r.t. the interface dispalcement jump increment. See
+  /// PureElasticPK1 for an example
+  virtual void computeTractionIncrementAndDerivatives() = 0;
+
+  virtual void initQpStatefulProperties() override;
+  virtual void computeQpProperties() override;
+
+  /// normal to the interface
```

Comment:
Do we document somewhere if this is inward or outward facing?

Reply:
this is the outward normal to the master surface in the undeformed configuration, I'll add this to the comment